### PR TITLE
Fix few docs tickets

### PR DESCRIPTION
**Pull request by tapaswenipathak**
ref https://github.com/mattermost/docs/pull/2732#issuecomment-502707886.

### CODE REVIEW

**Comment by amyblais** [comment]
@icelander Part of this is for https://github.com/mattermost/docs/issues/2231.

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -361,6 +361,16 @@ A 32-character key for encrypting and decrypting sensitive fields in the databas
...
 
+The following fields are encrypted using this key
+
+- ``SqlSettings.DriverName``
+- ``SqlSettings.DataSource``
+- ``SqlSettings.MaxIdleConns``
+- ``SqlSettings.MaxOpenConns``
+- ``SqlSettings.Trace``
+- ``SqlSettings.QueryTimeout``
+- ``SqlSettings.ConnMaxLifetimeMilliseconds``
```
Comment (jwilander) [inline source/administration/config-settings.rst]: I'm pretty sure this isn't true. I don't think the AtRestEncrypt key is used for anything at the moment and is a legacy setting. I'm curious as to how you came up with that?

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: I found `AtRestEncrypt` https://github.com/search?p=2&q=org%3Amattermost+SqlSettings&type=Code being set at quite a few places, I think the fields of `SqlSettings` are being encrypted using `AtRestEncrypt`, right?

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: See this https://github.com/search?q=org%3Amattermost+SqlSettings.AtRestEncryptKey&type=Code

Comment (jwilander) [reply source/administration/config-settings.rst]: Sure there are reference to it in code but it's not being used for any actual encrypting. Can you point to where it's actually doing what you say?

>The following fields are encrypted using this key

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: I looked at quite a lot of things, I don't remember now, correct if I'm wrong.
- https://github.com/mattermost/mattermost-server/commit/34d688ca72d8ef13b83987cc3b3d9b93ab989d9f
- https://github.com/mattermost/docs/issues/2231
- https://github.com/mattermost/mattermost-server/issues/9741
- https://github.com/mattermost/mattermost-redhat-docker/blob/e9d707231cc867fcc265fdb3ebe4073c3e91a340/v5/entrypoint.sh#L52

I didn't find a direct reference of `AtRestEncryptKey`, what I understood was `AtRestEncryptKey` encrypts anything `SqlSettings`. I agree that there is no need for encrypting all fields of `SqlSettings`. Why do you have `AtRestEncryptKey` then? Just curious.

Comment (jwilander) [reply source/administration/config-settings.rst]: It's a legacy setting that we used to use for encrypting data being stored at rest in the database. The setting was never really fully used and we just haven't removed it

Comment (jwilander) [reply source/administration/config-settings.rst]: If you want to remove this section from the PR the other parts look good

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: I added a31109e, looks ok? or should I remove this line? I suggest you create a ticket for this.

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: Does a31109e, looks ok? or should I remove the line? I suggest you create a ticket.

Comment (jwilander) [reply source/administration/config-settings.rst]: That's fine, thanks!

File: source/administration/config-settings.rst [hunk 2]
```diff
@@ -1146,6 +1149,18 @@ After changing this setting, the setting will take effect after the next time th
...
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
+Session length for SSO authentication (days)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ This setting defines the session length for SSO authentication, such as GitLab and SAML.
+
+ Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML or GitLab, the user may automatically be logged back in to Mattermost if they are already logged in to SAML or GitLab.
+
+ After changing this setting, the setting will take effect after the next time the user enters their credentials.
```
Comment (jwilander) [inline source/administration/config-settings.rst]:
```suggestion
After changing this setting, the setting will take effect after the next time the user enters their credentials.
```

File: source/administration/config-settings.rst [hunk 3]
```diff
@@ -361,6 +361,9 @@ A 32-character key for encrypting and decrypting sensitive fields in the databas
 
 When using High Availability, the salt must be identical in each instance of Mattermost.
 
+No fields are encrypted using `AtRestEncryptKey`. It's a legacy setting used to
+use for encrypting data being stored at rest in the database.
```
Comment (jwilander) [inline source/administration/config-settings.rst]:
```suggestion
encrypt data being stored at rest in the database.
```

File: source/administration/config-settings.rst [hunk 4]
```diff
@@ -1146,6 +1149,18 @@ After changing this setting, the setting will take effect after the next time th
 | This feature's ``config.json`` setting is ``"SessionLengthSSOInDays" : 30`` with whole number input.                                                                 |
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
+Session length for SSO authentication (days)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ This setting defines the session length for SSO authentication, such as GitLab and SAML.
```
Comment (jwilander) [inline source/administration/config-settings.rst]:
```suggestion
This setting defines the session length for SSO authentication, such as GitLab and SAML.
```

File: source/administration/config-settings.rst [hunk 5]
```diff
@@ -1146,6 +1149,18 @@ After changing this setting, the setting will take effect after the next time th
...
+
+ Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML or GitLab, the user may automatically be logged back in to Mattermost if they are already logged in to SAML or GitLab.
```
Comment (jwilander) [inline source/administration/config-settings.rst]:
```suggestion
Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML, GitLab or OAuth 2.0, the user may automatically be logged back in to Mattermost if they are already logged in to SAML, GitLab or with OAuth 2.0.
```

File: source/administration/config-settings.rst [hunk 6]
```diff
@@ -3078,6 +3093,18 @@ Specify the color of the AD/LDAP login button for white labeling purposes. Use a
...
+^^^^^^^^^^^^^^^^^^^^^^
+
+SAML session length is defined per
+
+ - ``url`` : ``environment/session_lengths``
+- ``title`` : ``admin.sidebar.sessionLengths``
+- ``type`` : ``TYPE_NUMBER``
+- ``label`` : ``mobileSessionDays``
+- ``help_text`` : ``admin.service.ssoSessionDaysDesc``
+- ``placeholder``: ``sessionIdleTimeout``
```
Comment (jwilander) [inline source/administration/config-settings.rst]: I'm not sure the above makes sense? How does that relate to SAML session length

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: We had a long discussion and multiple iteration on https://github.com/mattermost/docs/pull/2732, should I remove `placeholder`?

Comment (jwilander) [reply source/administration/config-settings.rst]: The content of this section doesn't make any sense to me. I'm not sure what it's trying to communicate. What does any of it mean? For example, what does the below communicate to a reader of this document?

```
``help_text`` : ``admin.service.ssoSessionDaysDesc``
```

Comment (tapaswenipathak) [reply source/administration/config-settings.rst]: The point is, I don't remember now. I have deleted these lines - fa3495e.

File: source/administration/config-settings.rst [hunk 7]
```diff
@@ -1146,6 +1149,18 @@ After changing this setting, the setting will take effect after the next time th
 | This feature's ``config.json`` setting is ``"SessionLengthSSOInDays" : 30`` with whole number input.                                                                 |
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
+Session length for SSO authentication (days)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This setting defines the session length for SSO authentication, such as GitLab and SAML.
```
Comment (jwilander) [inline source/administration/config-settings.rst]:
```suggestion
This setting defines the session length for SSO authentication, such as SAML, GitLab and OAuth 2.0.
```

File: source/administration/config-settings.rst [hunk 8]
```diff
@@ -361,6 +361,9 @@ A 32-character key for encrypting and decrypting sensitive fields in the databas
 
 When using High Availability, the salt must be identical in each instance of Mattermost.
 
+No fields are encrypted using `AtRestEncryptKey`. It's a legacy setting used to
```
Comment (amyblais) [inline source/administration/config-settings.rst]:
```suggestion
No fields are encrypted using ``AtRestEncryptKey``. It's a legacy setting used to
```

File: source/administration/config-settings.rst [hunk 9]
```diff
@@ -361,6 +361,9 @@ A 32-character key for encrypting and decrypting sensitive fields in the databas
...
+encrypt data being stored at rest in the database.
```
Comment (amyblais) [inline source/administration/config-settings.rst]:
```suggestion
encrypt data stored at rest in the database.
```

File: source/administration/config-settings.rst [hunk 10]
```diff
@@ -1146,6 +1149,18 @@ After changing this setting, the setting will take effect after the next time th
 | This feature's ``config.json`` setting is ``"SessionLengthSSOInDays" : 30`` with whole number input.                                                                 |
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
+Session length for SSO authentication (days)
```
Comment (amyblais) [inline source/administration/config-settings.rst]: I think this setting is accidentally repeated twice here.

File: source/administration/config-settings.rst [hunk 11]
```diff
@@ -1138,10 +1141,12 @@ After changing this setting, the new session length will take effect after the n
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML or GitLab, the user may automatically be logged back in to Mattermost if they are already logged in to SAML or GitLab.
 
-After changing this setting, the setting will take effect after the next time the user enters their credentials.
+This setting defines the session length for SSO authentication, such as SAML, GitLab and OAuth 2.0.
 
+Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML, GitLab or OAuth 2.0, the user may automatically be logged back in to Mattermost if they are already logged in to SAML, GitLab or with OAuth 2.0.
+
+After changing this setting, the setting will take effect after the next time the user enters their credentials.
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
Comment (amyblais) [inline source/administration/config-settings.rst]: Please help fix formatting of the table.

**Pull request merged**